pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -244,6 +243,62 @@ public Response downloadSegment(
     }
   }
 
+  // Upload a low level consumer segment to segment store and return the segment download url
```

Comment:
Add a pointer to the design section here, so that there is some context:
https://cwiki.apache.org/confluence/display/PINOT/By-passing+deep-store+requirement+for+Realtime+segment+completion#BypassingdeepstorerequirementforRealtimesegmentcompletion-Failurecasesandhandling
Mention in comments that this API is used when segment store is unavailable during llc completion.

Reply:
Also mention here that the invocation of this API may cause query performance to suffer, since we tar up the segment to upload it. 
You may also want to add a comment that upload failure is ok here, since the caller may choose to retry later

Reply:
Thanks for the pointer @mcvsubbu .  Comments updated.